### PR TITLE
Add Conditions list search

### LINKED ISSUE #82: [FE] - Condition List - Search
- Add search component
- Filter results base on search input
- What columns to search in???? **TBD**

**Wireframe:**
https://wireframepro.mockflow.com/view/M6eaccdb4e3776bf4fd576d04b0095a191590507310129#/page/D7f258f07dad136a6a9c600702937eb50

### PR DESCRIPTION
**Pull request by davidalbela**
- [x] Create gql `ConditionsSearchQuery` to filter by oracle address and orderBy `resolveBlockNumber`.
- [x] Add search conditions by oracle address.

Closes #82

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr98--conditionaltokensexplorer.review.gnosisdev.com

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr98--conditionaltokensexplorer.review.gnosisdev.com

File: src/pages/ConditionsList/index.tsx [hunk 1]
```diff
@@ -53,8 +53,17 @@ const customStyles = {
 }
 
 export const ConditionsList = () => {
-  const { data, error, loading } = useQuery<Conditions>(ConditionsListQuery)
+  const [searchTerm, setSearchTerm] = React.useState('')
+  const { data, error, loading } = useQuery<Conditions>(
+    searchTerm ? ConditionsSearchQuery : ConditionsListQuery,
+    {
+      variables: { conditionId: searchTerm },
```
Comment (mariano-aguero) [inline src/pages/ConditionsList/index.tsx:60]: I saw the design, I founded the only property that we need to use to search, is the conditionId (see placeholder). That said , I think the _searchTerm_ property must be called (maybe) _conditionIdToSearch_ or something more clear to understand that the variable is related to "conditionIds"
Screenshot_20200820_165428

File: src/queries/conditions.tsx [hunk 2]
```diff
@@ -1,8 +1,30 @@
...
+      creator
+      resolveBlockNumber
+    }
+  }
+`
+
 export const ConditionsListQuery = gql`
   query Conditions {
-    conditions(first: 1000) {
+    conditions(first: 1000, orderBy: $DEFAULT_ORDER_FIELD, orderDirection: desc) {
```
Comment (mariano-aguero) [inline src/queries/conditions.tsx:27]: The variable `DEFAULT_ORDER_FIELD` I think is not needed, and is not passed as a variables in the useQuery.
I dont know what exactly this variable does.
We can see in the image the variable is not used, any thoughts about it @hpmaxi @lmcorbalan ?
Screenshot_20200820_170544

File: src/queries/conditions.tsx [hunk 3]
```diff
@@ -1,5 +1,19 @@
 import gql from 'graphql-tag'
 
+export const ConditionsSearchQuery = gql`
+  query Conditions($conditionId: String!) {
+    conditions(first: 1000, where: { id: $conditionId }) {
```
[the remark this answers is not in the document]
Comment (lmcorbalan) [reply src/queries/conditions.tsx:5]: Actually, as TheGraph returns at most 1000 elements, the best way would be to have a positions counter in the subgraph and to use that counter to create multiples queries to get the complete set of positions.

[the remark this answers is not in the document]
Comment (davidalbela) [reply src/queries/conditions.tsx:5]: Same reply as https://github.com/gnosis/conditional-tokens-explorer/pull/116/files#r475539215

File: src/pages/ConditionsList/index.tsx [hunk 4]
```diff
@@ -53,8 +53,17 @@ const customStyles = {
...
-  const { data, error, loading } = useQuery<Conditions>(ConditionsListQuery)
+  const [conditionIdToSearch, setConditionIdToSearch] = React.useState('')
+  const { data, error, loading } = useQuery<Conditions>(
+    conditionIdToSearch ? ConditionsSearchQuery : ConditionsListQuery,
+    {
+      variables: { conditionId: conditionIdToSearch },
+    }
+  )
   const history = useHistory()
+  const handleChange = (event: { target: { value: React.SetStateAction<string> } }) => {
```
[the remark this answers is not in the document]
Comment (mariano-aguero) [reply src/pages/ConditionsList/index.tsx]: @lmcorbalan What do you think to do this in another issue ? I think the PR is open for a while, and this is a blocker

File: src/queries/conditions.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (mariano-aguero) [reply src/queries/conditions.tsx:5]: @lmcorbalan Maybe we can work on this in a new issue

File: src/queries/conditions.tsx [hunk 5]
```diff
@@ -1,5 +1,19 @@
...
+      outcomeSlotCount
+      resolved
+      creator
+      resolveBlockNumber
+    }
+  }
+`
+
 export const ConditionsListQuery = gql`
   query Conditions {
```
[the remark this answers is not in the document]
Comment (davidalbela) [reply src/queries/conditions.tsx]: Done

File: src/queries/conditions.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (mariano-aguero) [reply src/queries/conditions.tsx:5]: Issue https://github.com/gnosis/conditional-tokens-explorer/issues/123

File: src/pages/ConditionsList/index.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (mariano-aguero) [reply src/pages/ConditionsList/index.tsx]: Issue https://github.com/gnosis/conditional-tokens-explorer/issues/122

**Comment by mariano-aguero** [comment]
Great job @davidalbela please solve the conflicts, and you are free to merge when you finish

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr98--conditionaltokensexplorer.review.gnosisdev.com

**Pull request merged**